tests/test_stdext.ml

```diff
@@ -0,0 +1,37 @@
+let is_backup_file () =
+  let check ~input ~expected =
+    let name = "Fpath.is_backup_file " ^ input in
+    let actual = Dune_release.Stdext.Fpath.is_backup_file input in
+    Alcotest.(check bool) name expected actual
+  in
+  check ~input:"" ~expected:false;
+  check ~input:"fooooooooooooo" ~expected:false;
+  check ~input:"fooooooooo#" ~expected:false;
+  check ~input:"#fooooooooo#" ~expected:true;
+  check ~input:"foooooooooooo~" ~expected:true
+
+let find_files () =
+  let check ~path ~name_wo_ext ~expected =
+    let name = "Fpath.find_files " ^ path in
+    let actual =
+      Dune_release.Stdext.Fpath.find_files ~name_wo_ext (Fpath.v path)
+    in
+    let open Alcotest in
+    let open Alcotest_ext in
+    (check (result_msg (list path))) name expected actual
+  in
+  check ~path:"/should-not#exist" ~name_wo_ext:""
+    ~expected:
+      (Error
+         (`Msg
+           "directory contents /should-not#exist: No such file or directory"));
```

Comment:
These tests seem a bit fragile. I'd rather not have the unit test depend on the file system in that way.

My suggestion to make this function pure allows to test the logic only and not to test `OS.Dir.content`, which is what this first test case is about.

Reply:
also done